doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,153 @@
+more explanations. For a TL;DR, jump to the :ref:`Complete configuration`.
+
+Install the Haskell environment
+-------------------------------
+
+Haskell development on Windows makes use of the `MSYS2 <https://www.msys2.org/>`_ 
+tools.
+
+The recommended way of setting up a Haskell environment in Windows is by using
+`GHCup <https://www.haskell.org/ghcup/>`_. Follow the steps outlined in its
+webpage to install at least GHC and Cabal. GHCup will install its own MSYS2
+system in your computer unless told not to do so: refer to `its documentation
+<https://www.haskell.org/ghcup/install/#windows_1>`_ for more information.
+
+.. NOTE::
+   Stack is another tool you can use to set up a Haskell environment on Windows. Stack
+   can be installed on its own or via GHCup. See 
+   `Stack's webpage <https://docs.haskellstack.org/en/stable/>`_ and/or 
+   `GHCup's section on Stack integration <https://www.haskell.org/ghcup/guide/#stack-integration>`_,
+   in particular the `Windows related subsection <https://www.haskell.org/ghcup/guide/#windows>`_.
+
+Ensure that Cabal can call the tools it needs
+---------------------------------------------
+
+Cabal sometimes needs to call tools that do not come with Windows (such as
+``make`` or even ``git``). The MSYS2 project makes many of them available on
+Windows. The directories where those are located need to be made visible in the
+``PATH`` when executing ``cabal``. For that, Cabal provides the
+``extra-prog-path`` configuration option. Your :ref:`global configuration
+<config-file-discovery>` should include this option:
+
+::
+
+   extra-prog-path: <msys-dir>\<environment>\bin
+                    <msys-dir>\usr\bin
+
+Where ``<msys-dir>`` points to the location of your MSYS2 installation. Refer to
+GHCup's documentation on the default location of this directory.
+``<environment>`` has to be one of the environments of MSYS2, which for GHCup is
+``mingw64``. You can learn more about the different environments in the `MSYS2
+documentation <https://www.msys2.org/docs/environments/>`_.
+
+.. note::
+
+   Unless told otherwise, GHCup already adds these directories to `extra-prog-path`
+   by default.
+
+Ensure that Cabal can use system libraries
+------------------------------------------
+
+Third-party libraries can be installed using the ``pacman`` package manager on
+the MSYS2 installation. When installing a third party package its libraries and
+header files will (usually) be placed in
+``<msys-dir>\<environment>\{lib,include}`` respectively. These directories need
+to be specified in the ``extra-lib-dirs`` and ``extra-include-dirs``
+respectively. Your :ref:`global configuration <config-file-discovery>` should
+include these options:
+
+::
+
+   extra-include-dirs: <msys-dir>\<environment>\include
+   extra-lib-dirs: <msys-dir>\<environment>\lib
+
+
+.. note::
+
+   Unless told otherwise, GHCup already adds these directories to `extra-include-dirs` and `extra-lib-dirs`
+   by default.

```

Comment:
The ghcup bootstrap script does. Not ghcup itself. That's an important difference.

Reply:
Ah, shall I say "When bootstrapping GHCup..."?

(side note, I see you use `ghcup` not `GHCup`. Do you have a preference that I change mentions to `ghcup` instead?)

Reply:
'GHCup' is the project, `ghcup` usually refers to the binary. The bootstrap script bootstraps the binary. On windows, it also installs msys2 (the powershell script) and changes the cabal config (the bash script, which is invoked by the powershell script and is the same that you run on linux) as you pointed out. The info doesn't need to be that detailed, but should refer to the bootstrap scripts.

Reply:
Done :+1: